api/v1beta1/cinderapi_types.go

```diff
@@ -23,8 +23,36 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+
+// CinderAPITemplate defines the input parameters for the Cinder API service
+type CinderAPITemplateCore struct {
+	// Common input parameters for the Cinder API service
+	CinderServiceTemplate `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// Replicas - Cinder API Replicas
+	Replicas *int32 `json:"replicas"`
+
+	// +kubebuilder:validation:Optional
+	// Override, provides the ability to override the generated manifest of several child resources.
+	Override APIOverrideSpec `json:"override,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// TLS - Parameters related to the TLS
+	TLS tls.API `json:"tls,omitempty"`
+}
+
+
 // CinderAPITemplate defines the input parameters for the Cinder API service
 type CinderAPITemplate struct {
+
+    // +kubebuilder:validation:Required
+    // ContainerImage - Cinder Container Image URL (will be set to environmental default if empty)
+    ContainerImage string `json:"containerImage"`
+
 	// Common input parameters for the Cinder API service
 	CinderServiceTemplate `json:",inline"`
```

Comment:
Why don't we use `CinderAPITemplateCore` instead?

Because L29-L45 are the same as L58-L72.

Reply:
fixed all of these in the latest rebase. @Akrog thanks!